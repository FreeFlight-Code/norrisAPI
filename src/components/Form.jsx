import React from 'react';

class Form extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			email: "",
			password: ""
		};
		this.handleSubmit = this.handleSubmit.bind(this);
		this.handleInputChange = this.handleInputChange.bind(this);
		this.togglePasswordvisibility = this.togglePasswordvisibility.bind(this);
	}



	validateEmailAndPassword(email, password) {
		//validation checks needed
		return true
	}

	togglePasswordvisibility() {
		var x = document.getElementById("password");
		if (x.type === "password") {
			x.type = "text";
		} else {
			x.type = "password";
		}
	}

	handleSubmit(e) {
		e.preventDefault();
		const { email, password } = this.state;
		const user = {
			email: email,
			password: password
		}
		if (this.validateEmailAndPassword(email, password)) {
			this.props.loginUser(user);
			this.setState({
				email: "",
				password: ""
			})
		}
		
	}

	handleInputChange(e) {
		const { type, value } = e.target;
		this.setState({
			[type]: value
		});
	}

	render() {
	const { email, password } = this.state;	

	return (
		<form className="loginForm">
			<label>Email</label>
			<input
				onChange={e => this.handleInputChange(e)}
				// pattern="^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$"
				type="email"
				name="email"
				id="email"
				value={email}
			/>
			<span className="passwordContainer">
				<label>Password</label>
				<input
					onChange={_ => this.togglePasswordvisibility()}
					type="checkbox"
				/>
			</span>
			<input
				onChange={e => this.handleInputChange(e)}
				// pattern="(?=.{8,})"
				type="password"
				name="password"
				id="password"
				value={password}
			/>
			<button
				onClick={e => this.handleSubmit(e)}
				className="blue"
			>
				Submit
			</button>
		</form>
	);
  }
}
 
export default Form;