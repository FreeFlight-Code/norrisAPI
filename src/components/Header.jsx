import React from 'react';
import {BrowserRouter as Router, Link} from 'react-router-dom';

class Header extends React.Component {
	constructor(props) {
		super(props);
		this.state = {  }
	}


	render() { 
		return (
      <header>
        <nav>
			<a href="/"><span>Home</span></a>
			<a href="jokes"><span>Jokes</span></a>
			<a href="categories"><span>Categories</span></a>
			<a href="search"><span>Search</span></a>
        </nav>
      </header>
    );
	}
}
 
export default Header;