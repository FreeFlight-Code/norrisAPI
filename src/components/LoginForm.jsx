import React from "react";
import { connect } from "react-redux";

class LoginForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      password: ""
    };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);
    this.togglePasswordvisibility = this.togglePasswordvisibility.bind(this);
    // this.validateEmailAndPassword = this.validateEmailAndPassword.bind(this);
    document.addEventListener("keyup", this.validateEmailAndPassword);
  }

  componentWillUnmount() {
    document.removeEventListener("keyup", this.validateEmailAndPassword);
  }

  validateEmailAndPassword = () => {
    const emailRegex = new RegExp(
      "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+.[a-zA-Z]{2,}$"
    );
    const passwordRegex = new RegExp("^[a-zA-Z0-9]{6,10}$");
    const { email, password } = this.state;
    const submitButton = document.querySelector("button.submit");
    if (emailRegex.test(email) && passwordRegex.test(password)) {
      if (submitButton.className.indexOf("disabled") > 0) {
        submitButton.classList.remove("disabled");
      }
    } else {
      if (submitButton.className.indexOf("disabled") < 0) {
        submitButton.classList.add("disabled");
      }
    }
  };

  togglePasswordvisibility() {
    var x = document.getElementById("password");
    if (x.type === "password") {
      x.type = "text";
    } else {
      x.type = "password";
    }
  }

  handleSubmit(e) {
    e.preventDefault();
    const { email, password } = this.state;
    this.props.dispatch({ type: "DISPLAY_MODAL" });
    this.props.dispatch({ type: "LOG_IN", password: password, email: email });
    this.setState({
      email: "",
      password: ""
    });
    setTimeout(() => {
      this.props.dispatch({ type: "REMOVE_MODAL" });
    }, 3000);
  }

  handleInputChange({ type, value }) {
    this.setState({
      [type]: value
    });
  }

  render() {
    const { email, password } = this.state;

    return (
      <form className="loginForm">
        <label>Email</label>
        <input
          onChange={e => this.handleInputChange(e.target)}
          type="email"
          name="email"
          id="email"
          value={email}
        />
        <span className="passwordContainer">
          <label>Password</label>
          <input
            onChange={_ => this.togglePasswordvisibility()}
            type="checkbox"
          />
        </span>
        <input
          onChange={e => this.handleInputChange(e.target)}
          type="password"
          name="password"
          id="password"
          value={password}
        />
        <button
          onClick={e => this.handleSubmit(e)}
          className="submit blue glow disabled"
        >
          Submit
        </button>
      </form>
    );
  }
}

export default connect()(LoginForm);
