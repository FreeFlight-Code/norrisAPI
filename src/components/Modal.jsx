import React from "react";
import {connect} from 'react-redux';
import {removeModal} from '../redux/modal';

export function Modal (props) {
	const {message, messageType, show} = props.modal;

	if(show){
		return (
			<div className="modal" onClick={e=>{props.dispatch(removeModal());}}>
				<div className={messageType}>
					{message ? message :
					<img alt="animated loading gif" src="https://i.pinimg.com/originals/a2/dc/96/a2dc9668f2cf170fe3efeb263128b0e7.gif"/>
					}
				</div>
			</div>
		);

	} else {
		return null
	}
}


const mapStateToProps = state => {
  return {
    modal: state.modal
  };
};

export default connect(mapStateToProps)(Modal);
