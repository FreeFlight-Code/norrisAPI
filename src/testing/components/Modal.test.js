import React from 'react';
import {Modal} from '../../components/Modal';
import renderer from 'react-test-renderer';

const children = {
  message: "",
  messageType: "",
  show: false
};
describe('the modal component', ()=>{
	it('renders', () => {
		const tree = renderer
			.create( < Modal modal={children} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

})
