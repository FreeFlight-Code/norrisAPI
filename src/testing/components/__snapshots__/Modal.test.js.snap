// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the modal component renders 1`] = `null`;
